vector/src/main/java/im/vector/app/features/settings/push/PushGatewaysViewModel.kt

```diff
@@ -70,10 +72,17 @@ class PushGatewaysViewModel @AssistedInject constructor(@Assisted initialState:
 
     override fun handle(action: PushGatewayAction) {
         when (action) {
-            is PushGatewayAction.Refresh -> handleRefresh()
+            is PushGatewayAction.Refresh      -> handleRefresh()
+            is PushGatewayAction.RemovePusher -> removePusher(action.pusher)
         }.exhaustive
     }
 
+    private fun removePusher(pusher: Pusher) {
+        viewModelScope.launch {
+            session.removePusher(pusher.pushKey, pusher.appId)
```

Comment:
You have to protect this call with a try catch. If triggered in Airplane mode, it will crash the app. In case of Failure you can use ViewEvent to sent the failure and display it in a dialog (many examples exist in the code)

Reply:
good point, will fix :+1: I had gotten used to debug pages taking shortcuts :sweat_smile:

Reply:
4c4f2fce74d3b0f0dc4fe6136b40b9e440e90116

![offline-remove-pusher](https://user-images.githubusercontent.com/1848238/134502910-a5c46361-5872-4856-9084-db407292e8b4.gif)

Reply:
Thanks, crash in debug pages are also bad for the statistics :)